src/aurora/actionscript/variable.h

```diff
@@ -66,6 +66,8 @@ class Variable {
 	const Common::UString &asString();
 	bool asBoolean();
 
+	template<typename T> boost::shared_ptr<T> as();
```

Comment:
You're using that functionality in 8b02e03dc761c51efc53bc63f3794899142e779e, which is *before* this commit.

Please order your commits so that every single commit still compiles. I.e. order this commit *before* the "Add basic array class" one.

This is really crucial for looking for regression with "git bisect". When there's a commit in the middle that doesn't compile, that doesn't exactly break the bisect, but it adds annoyances.